pkg/network/tracer/tracer_classification_test.go

```diff
@@ -0,0 +1,278 @@
+)
+
+func testProtocolClassification(t *testing.T, cfg *config.Config, clientHost, targetHost, serverHost string) {
+	dialer := &net.Dialer{
+		LocalAddr: &net.TCPAddr{
+			IP:   net.ParseIP(clientHost),
+			Port: 0,
+		},
+	}
+	tests := []struct {
+		name      string
+		clientRun func(t *testing.T, serverAddr string)
+		serverRun func(t *testing.T, serverAddr string, done chan struct{}) string
+		want      network.ProtocolType
+	}{
+		{
+			name: "udp client",
+			clientRun: func(t *testing.T, serverAddr string) {
+				c, err := net.DialTimeout("udp", serverAddr, time.Second)
+				require.NoError(t, err)
+				defer c.Close()
+
+				for i := 0; i < 5; i++ {
+					_, err = c.Write(genPayload(clientMessageSize))
+					require.NoError(t, err)
+
+					buf := make([]byte, serverMessageSize)
+					_, err = c.Read(buf)
+					require.NoError(t, err)
+				}
+			},
+			serverRun: func(t *testing.T, serverAddr string, done chan struct{}) string {
+				udpServer := &UDPServer{
+					address: serverAddr,
+					onMessage: func(b []byte, n int) []byte {
+						return genPayload(serverMessageSize)
+					},
+				}
+				require.NoError(t, udpServer.Run(done, clientMessageSize))
+				return udpServer.address
+			},
+			want: network.ProtocolUnclassified,
+		},
+		{
+			name: "tcp client without sending data",
+			clientRun: func(t *testing.T, serverAddr string) {
+				timedContext, cancel := context.WithTimeout(context.Background(), time.Second)
+				c, err := dialer.DialContext(timedContext, "tcp", serverAddr)
+				cancel()
+				require.NoError(t, err)
+				defer c.Close()
+			},
+			serverRun: func(t *testing.T, serverAddr string, done chan struct{}) string {
+				server := NewTCPServerOnAddress(serverAddr, func(c net.Conn) {
+					c.Close()
+				})
+				require.NoError(t, server.Run(done))
+				return server.address
+			},
+			want: network.ProtocolUnclassified,
+		},
+		{
+			name: "tcp client with sending random data",
+			clientRun: func(t *testing.T, serverAddr string) {
+				timedContext, cancel := context.WithTimeout(context.Background(), time.Second)
+				c, err := dialer.DialContext(timedContext, "tcp", serverAddr)
+				cancel()
+				require.NoError(t, err)
+				defer c.Close()
+				c.Write([]byte("hello\n"))
+			},
+			serverRun: func(t *testing.T, serverAddr string, done chan struct{}) string {
+				server := NewTCPServerOnAddress(serverAddr, func(c net.Conn) {
+					r := bufio.NewReader(c)
+					input, err := r.ReadBytes(byte('\n'))
+					if err == nil {
+						c.Write(input)
+					}
+					c.Close()
+				})
+				require.NoError(t, server.Run(done))
+				return server.address
+			},
+			want: network.ProtocolUnknown,
+		},
+		{
+			name: "tcp client with sending HTTP request",
+			clientRun: func(t *testing.T, serverAddr string) {
+				client := nethttp.Client{
+					Transport: &nethttp.Transport{
+						DialContext: dialer.DialContext,
+					},
+				}
+				resp, err := client.Get("http://" + serverAddr + "/test")
```

Comment:
I think it'd make the tests more robust if we sent each type of HTTP request that we can classify, instead of just `GET`.

Reply:
@richl-dd perhaps... but that's beyond scope of this PR.  This is incorporating the existing tests.